app/assets/javascripts/discourse/app/controllers/composer.js

```diff
@@ -289,12 +290,18 @@ export default Controller.extend({
     return SAVE_LABELS[modelAction];
   },
 
-  @discourseComputed("isStaffUser", "model.action")
-  canWhisper(isStaffUser, modelAction) {
+  @discourseComputed("userGroups", "model.action")
+  canWhisper(userGroups, modelAction) {
+    if (isEmpty(this.siteSettings.enable_whispers)) {
+      return false;
+    }
+    const whispererGroups = this.siteSettings.enable_whispers
+      .split("|")
+      .map(Number);
     return (
-      this.siteSettings.enable_whispers &&
-      isStaffUser &&
-      Composer.REPLY === modelAction
+      Composer.REPLY === modelAction &&
+      userGroups &&
```

Comment:
I believe this will be an empty array if the user does not belong to any group?

Reply:
I also wonder if we can just return earlier in the function if the user does not belong to any group